hud/src/androidLibMain/kotlin/AndroidHUD.kt

```diff
@@ -76,14 +94,20 @@ class AndroidHUD private constructor(viewResId: Int, hudConfig: HudConfig, priva
             private const val STYLE_KEY = "style"
             private const val TITLE_KEY = "title"
 
-            fun newInstance(@IdRes viewResId: Int, hudConfig: HudConfig) = LoadingDialog().apply {
+            fun newInstance(@LayoutRes viewResId: Int, hudConfig: HudConfig) = LoadingDialog().apply {
                 arguments = Bundle().apply {
                     putInt(RESOURCE_ID_KEY, viewResId)
                     putInt(STYLE_KEY, hudConfig.style.ordinal)
                     putString(TITLE_KEY, hudConfig.title)
                 }
                 isCancelable = false
-                retainInstance = true
+            }
+        }
+
+        override fun onCreate(savedInstanceState: Bundle?) {
+            super.onCreate(savedInstanceState)
+            if (savedInstanceState != null) {
```

Comment:
This is a curious line. It should at least have a comment to explain why we do this, but I'm not sure we want to either.

Reply:
Well, we relay on dialog state live data to show/dismiss. On rotation dialog fragment will be recreated to new instance, but don't want this, because we will show it again inside live data observe during the rotation process

Reply:
I'll add comments to explain